test/auto_parallel/spmd_rules/test_elementwise_rule.py

```diff
@@ -559,6 +559,108 @@ def test_backward_multi_mesh_dim_broadcast(self):
             infered_output_dist_attrs[0].dims_mapping, [-1, 0, -1, 1]
         )
 
+    def test_single_mesh_dim_greater_than(self):
+        binary_rule = core.get_phi_spmd_rule("greater_than")
+        # [0, -1], [-1, -1] --> [0, -1], [0, -1], [0, -1]
+        self.x_dist_tensor_spec.set_dims_mapping([0, -1])
+        self.y_dist_tensor_spec.set_dims_mapping([-1, -1])
+        result_dist_attrs = binary_rule.infer_forward(
+            self.x_dist_tensor_spec, self.y_dist_tensor_spec
+        )
+        infered_input_dist_attrs = result_dist_attrs[0]
+        infered_output_dist_attrs = result_dist_attrs[1]
+
```

Comment:
before check value of dims_mapping of input and output, should check len(infered_input_dist_attrs) and len(infered_output_dist_attrs) first